### ISSUE TITLE
Selected Video Plays Different Video

### ISSUE THREAD

**Issue by willpower68** [WWBN/AVideo#3125]
Has anyone experienced this issue.  When selecting a video from playlist. A different video loads.

<img width="869" alt="Screen Shot 2020-05-23 at 9 15 19 PM" src="https://user-images.githubusercontent.com/739391/82744730-7851f800-9d4a-11ea-9d7f-d75a9b47382f.png">

2nd Video: same behavior

<img width="939" alt="Screen Shot 2020-05-23 at 9 09 52 PM" src="https://user-images.githubusercontent.com/739391/82744739-b5b68580-9d4a-11ea-8dec-e7bf39be7ef7.png">

The thumb nail shows the video I selected to play but as you can see in player it's another video.

**Comment by DanielnetoDotCom** [comment]
I notice this problem in the past but should be fixed in the latest release

**Comment by willpower68** [comment]
@DanielnetoDotCom 
You game me an update about 3 week ago.  Is there more recent update?

**Comment by DanielnetoDotCom** [comment]
@willpower68 we have updates almost every day.

Please check this https://github.com/WWBN/AVideo/wiki/How-to-Update-your-AVideo-Platform

**Comment by willpower68** [comment]
@DanielnetoDotCom 
Hi Daniel, 

I hope you can give me a little more information to install the update.  I know that there is a command I am missing which is why I am not able to perform the update properly.

I'm logging into the root dir.  What are the commands to find the Avideo directory.  

Hope the images below will give insight into what I've done.

![git pull 1](https://user-images.githubusercontent.com/739391/82946763-f261c700-9f6c-11ea-8fcb-a8690f4ecbdf.JPG)


![git pull 2](https://user-images.githubusercontent.com/739391/82946780-f8f03e80-9f6c-11ea-8917-ff9c88acf6be.JPG)

**Comment by DanielnetoDotCom** [comment]
the command is only `sudo git pull`

with out the **&&**

**Comment by ronaldod** [comment]
locate first your home dir where the installation is installed.  Mostly with ubuntu the directory is located in the /var/www/ Then you go into that directory then run that `sudo git pull`

**Comment by willpower68** [comment]
@DanielnetoDotCom @ronaldod  thank you.
I am updating and hope I didn't break anything in the process.

**Comment by willpower68** [comment]
@DanielnetoDotCom 

I have updated the website but I still have issue of the when I select video A but video B plays.


<img width="1086" alt="Screen Shot 2020-05-27 at 6 53 04 AM" src="https://user-images.githubusercontent.com/739391/83010828-dbf85180-9fe6-11ea-879b-3f055660ba41.png">

**Comment by willpower68** [comment]
It seem the embed url ends up in Video Link field.  I deleted video link and hit save but the link is not removed.
 
<img width="1091" alt="Screen Shot 2020-05-27 at 1 17 32 PM" src="https://user-images.githubusercontent.com/739391/83052184-f2b99b00-a01c-11ea-9a9c-7624cc70a82c.png">

**Comment by DanielnetoDotCom** [comment]
@willpower68 is not clear what you are trying to do.

do you still have problems with the playlist order?
why do you want to remove the video link?

**Comment by willpower68** [comment]
Yes I am still having a problem.  

The video link (youtube link) is the video that is showing up.  I thought if I removed the link the correct video would play.  

if you choose 'Adriene Williams Daniel' video, the embeded youtube video of Jess Richard's..." will play and the link of that video is appearing in Video Link section of 'Adriene William Daniel'.  the Adriene video is not embeded.  It was uploaded video file.

**Comment by DanielnetoDotCom** [comment]
Hi

to remove the videoLink you must do it directly from the database.

the question is, how this video link appears there?

can you reproduce? it is probably a bug somewhere but we must reproduce it to be able to fix.

**Comment by willpower68** [comment]
@DanielnetoDotCom 
I will do my best to reproduce.  I have the same issue with 4 other videos.

**Comment by WGtechy** [comment]
i am also having same issue

**Comment by willpower68** [comment]
> i am also having same issue.  that's it exactly

**Comment by willpower68** [comment]
I'm not sure if this is related.  When I use 'site' command in google the results have the title and images aren't corresponding.  It seems to only occur with active videos and embed video.  In the example below, the name and picture is don't match.


<img width="612" alt="Screen Shot 2020-05-28 at 9 01 17 PM" src="https://user-images.githubusercontent.com/739391/83209478-7b702e00-a126-11ea-96b4-be64a940990d.png">

**Comment by akhilleusuggo** [comment]
I've talked with daniel about this time ago , we have some html description issues .
Also we need to implement <video:content_loc> on the sitemaps . If google can't download a portion of the video , will take longer to crawl the site ( maybe up to 2-3 months ) to get it right .

**Comment by WGtechy** [comment]
> > i am also having same issue.  that's it exactly

i need to confirm; did you delete the category that the videos were originally uploaded ?

**Comment by WGtechy** [comment]
> I've talked with daniel about this time ago , we have some html description issues .
> Also we need to implement video:content_loc on the sitemaps . If google can't download a portion of the video , will take longer to crawl the site ( maybe up to 2-3 months ) to get it right .

Okay its an issue that needs to be fixed.

**Comment by DanielnetoDotCom** [comment]
> I'm not sure if this is related. When I use 'site' command in google the results have the title and images aren't corresponding. It seems to only occur with active videos and embed video. In the example below, the name and picture is don't match.
> 
> <img alt="Screen Shot 2020-05-28 at 9 01 17 PM" width="612" src="https://user-images.githubusercontent.com/739391/83209478-7b702e00-a126-11ea-96b4-be64a940990d.png">

As I can see they are not the same video, looks like you uploaded twice and they have different posters 

![image](https://user-images.githubusercontent.com/18009293/83263991-d55c0c80-a195-11ea-9f2a-b21a25f03598.png)

**Comment by willpower68** [comment]
@DanielnetoDotCom 
Yes, I know they aren't the same video.  I have 4 videos with Adriene Williams as title.  

if you clicked on link that video would not be in the player. 

"...-3" link and image are not related and if you click on "..-4" another video will play

This issue regarding how they appear in search has been there since site launched.  

However, the issue with selecting a video from playlist and another video plays has been happening for about 3 or 4 weeks.   The videos played fine until 3 or 4 weeks ago.

**Comment by willpower68** [comment]
> > > i am also having same issue.  that's it exactly
> @WGtechy 
> i need to confirm; did you delete the category that the videos were originally uploaded ?

I did not delete any categories.  This issue started about 3 or 4 weeks ago.  Prior to that the videos played as expected.

**Comment by WGtechy** [comment]
> > > > i am also having same issue.  that's it exactly
> > > > @WGtechy
> > > > i need to confirm; did you delete the category that the videos were originally uploaded ?
> 
> I did not delete any categories. This issue started about 3 or 4 weeks ago. Prior to that the videos played as expected.

Im also facing exactly the same issue. I asked that question to see if the problem has something to do with categories. The affected files on my platform is more than 50 videos. If you check the tag section in the MY VIDEO dashboard, you will see that the type automatically changed from video to embed but still occupying space on the server. Embed videos are not suppose to take space since they are not hosted on the server. 

We hope there is a solution to this issue, some of my subscribers videos are affected.

**Comment by DanielnetoDotCom** [comment]
@willpower68 can you please share a link from a playlist with this error?

About the google indexing problem, it looks like Google is indexing the image from the first item from your left list.

![image](https://user-images.githubusercontent.com/18009293/83427246-cd56d380-a406-11ea-87ff-7df4b0ba970e.png)

I will see what I can do to decrease the relevance on it on the page, so google will not use that image anymore

**Comment by willpower68** [comment]
> @willpower68 can you please share a link from a playlist with this error?
> 
> About the google indexing problem, it looks like Google is indexing the image from the first item from your left list.
> 
> ![image](https://user-images.githubusercontent.com/18009293/83427246-cd56d380-a406-11ea-87ff-7df4b0ba970e.png)
> 
> I will see what I can do to decrease the relevance on it on the page, so google will not use that image anymore

@DanielnetoDotCom 
Here is link to the playlist with error:
https://memoriachannel.com/?search=adriene%20williams

**Comment by DanielnetoDotCom** [comment]
The provided link is not a playlist

**Comment by willpower68** [comment]
> The provided link is not a playlist
@DanielnetoDotCom 
My mistake
https://memoriachannel.com/plugin/PlayLists/player.php?playlists_id=3

**Comment by DanielnetoDotCom** [comment]
I guess something happens with the last video. (I guess that is the one you are talking about)

So the issue is not the playlist, it is the video itself. maybe the error on the last video is related to it https://github.com/WWBN/AVideo/issues/3156

**Comment by willpower68** [comment]
@DanielnetoDotCom 
Yes that's the video but it issue is with all her videos and few other that I uploaded.  I don't have any issue with embeded videos.

Here all the videos in one playlist.
https://memoriachannel.com/plugin/PlayLists/player.php?playlists_id=25

**Comment by DanielnetoDotCom** [comment]
@willpower68 sorry still not very clear for me...

the videos with errors in the playlist someone else upload it correct?

I am trying to see if the person who is uploading is doing something that rises a bug.

> Here all the videos in one playlist.
> https://memoriachannel.com/plugin/PlayLists/player.php?playlists_id=25
> missed a few videos
> https://memoriachannel.com/plugin/PlayLists/player.php?playlists_id=26

I do not understand the quoted above.

### One thing...

the playlist the main propose is for self-hosted videos, the ajax layout may have a hard time to use embed content and embed videos may not work properly.
try to check this option

![image](https://user-images.githubusercontent.com/18009293/83639680-690c4f00-a581-11ea-94e1-9cc2a8cc7b13.png)

**Comment by willpower68** [comment]
@DanielnetoDotCom 

thank you for the additional information regarding playlist.

The 

> @willpower68 sorry still not very clear for me...
> 
> the videos with errors in the playlist someone else upload it correct?
> 
> I am trying to see if the person who is uploading is doing something that rises a bug.
> 
> > Here all the videos in one playlist.
> > https://memoriachannel.com/plugin/PlayLists/player.php?playlists_id=25
> > missed a few videos
> > https://memoriachannel.com/plugin/PlayLists/player.php?playlists_id=26
> 
> I do not understand the quoted above.
> 
> ### One thing...
> the playlist the main propose is for self-hosted videos, the ajax layout may have a hard time to use embed content and embed videos may not work properly.
> try to check this option
> 
> ![image](https://user-images.githubusercontent.com/18009293/83639680-690c4f00-a581-11ea-94e1-9cc2a8cc7b13.png)

the video in the playlist are self-hosted, I uploaded those videos myself.  I had no issues with the video's a few weeks ago.

**Comment by willpower68** [comment]
@DanielnetoDotCom 
thank you for the additional about playlist

The video's are self-hosted.  I added the video's myself.

**Comment by willpower68** [comment]
@WGtechy I wanted to reach out because you mentioned you have paying members for you website.  Are you will to discuss your strategy for getting paying members to join.  Thanks

**Comment by DanielnetoDotCom** [comment]
But on the videos samples, you sent they are embedded.

maybe the problem I mention here https://github.com/WWBN/AVideo/issues/3125#issuecomment-637599214

So the videos are basically lost. and something that I could not detect yet. made this happen.

that is why I am asking a way to reproduce it. 

who is uploading the videos for that woman? you or her?

can you or here reproduce this error to have the self-hosted video replaced by an embed?

**Comment by leonidaswilliams** [comment]
Hi Daniel!

Your job with this tool is amazing! Thanks a lot.

I had the same problem and I found that when I created an LinkedVideo in the table videos, the link was applied to all the records, in the field videoLink. Also the field type were all of them as 'embed' and also the field externalOptions had the incorrect values for a video.

I modified all these fields and is working again!

Thanks a lot
